File: backend/src/main/java/com/containertrack/entity/ShippingCompany.java
package com.containertrack.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;

@Entity
@Table(name = "shipping_companies")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ShippingCompany {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(name = "short_code", nullable = false, unique = true, length = 20)
    private String shortCode;

    @Column(length = 100)
    private String country;

    @Column(name = "contact_email")
    private String contactEmail;

    @Column(name = "contact_phone", length = 50)
    private String contactPhone;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "free_days_limit", nullable = false)
    @Builder.Default
    private Integer freeDaysLimit = 10;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;
}

File: containertrack-frontend/src/types/shippingCompany.ts
export interface ShippingCompany {
  id: string;
  name: string;
  shortCode: string;
  country: string;
  contactEmail: string;
  contactPhone: string;
  notes: string;
  freeDaysLimit: number;
  isActive: boolean;
}

export interface ShippingCompanyRef {
  id: string;
  name: string;
  shortCode: string;
}

export interface CreateShippingCompanyRequest {
  name: string;
  shortCode: string;
  country: string;
  contactEmail: string;
  contactPhone: string;
  notes: string;
  freeDaysLimit: number;
}

export type UpdateShippingCompanyRequest = Partial<CreateShippingCompanyRequest>;

export interface UpdateShippingCompanyStatusRequest {
  isActive: boolean;
}
